src/data/real/golden_ratio.lean

```diff
@@ -0,0 +1,165 @@
+/-
+Copyright (c) 2020 Anatole Dedecker. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anatole Dedecker, Alexey Soloyev, Junyan Xu
+-/
+import data.real.irrational
+import data.nat.fib
+import data.matrix.notation
+import tactic.ring_exp
+import algebra.linear_recurrence
+
+/-!
+# The golden ratio and its conjugate
+
+This file defines the golden ratio `φ := (1 + √5)/2` and its conjugate
+`ψ := (1 - √5)/2`, which are the two real roots of `X² - X - 1`.
+
+Along with various computational facts about them, we prove their
+irrationality, and we link them to the Fibonacci sequence by proving
+Binet's formula.
+-/
+
+noncomputable theory
+
+/-- The golden ratio `φ := (1 + √5)/2`. -/
+@[reducible] def golden_ratio := (1 + real.sqrt 5)/2
+
+/-- The conjugate of the golden ratio `ψ := (1 - √5)/2`. -/
+@[reducible] def golden_conj := (1 - real.sqrt 5)/2
+
+localized "notation `φ` := golden_ratio" in real
+localized "notation `ψ` := golden_conj" in real
+
+/-- The inverse of the golden ratio is the opposite of its conjugate. -/
+@[simp] lemma inv_gold : φ⁻¹ = -ψ :=
```

Comment:
Could you add the "dual" simp lemma as well, please?

Reply:
Is `ψ⁻¹ = -φ` what you meant ?

Reply:
I'm not sure those are good simp lemmas. It's not clear that the RHS is simpler, it depends on context. Is there any `simp` call in this file using this?

Reply:
Nope

Reply:
Hmm, ok. Maybe the original was better than. In that case, the RHS was definitely simpler than LHS.
My reasoning is that in a field, getting rid of `¯¹` is usually a good thing, because you don't have to care about `≠ 0` issues.
That reminds me, probably we want some lemmas like `gold_ne_zero`, `gold_pos`, `one_lt_gold`, `gold_conj_lt_one`, `gold_conj_pos`, `gold_conj_ne_zero`.